pkg/apis/pipeline/v1alpha1/taskrun_validation.go

```diff
@@ -73,17 +73,23 @@ func (ts *TaskRunSpec) Validate() *apis.FieldError {
 }
 
 func (i TaskRunInputs) Validate(path string) *apis.FieldError {
-	if err := checkForPipelineResourceDuplicates(i.Resources, fmt.Sprintf("%s.Resources.Name", path)); err != nil {
+	if err := validatePipelineResources(i.Resources, fmt.Sprintf("%s.Resources.Name", path)); err != nil {
 		return err
 	}
 	return validateParameters(i.Params)
 }
 
 func (o TaskRunOutputs) Validate(path string) *apis.FieldError {
-	return checkForPipelineResourceDuplicates(o.Resources, fmt.Sprintf("%s.Resources.Name", path))
+	return validatePipelineResources(o.Resources, fmt.Sprintf("%s.Resources.Name", path))
 }
 
-func checkForPipelineResourceDuplicates(resources []TaskResourceBinding, path string) *apis.FieldError {
+/*
+validatePipelineResources validates that
+	* resource is not declared more than once
+	* if both resource reference and resource spec is defined at the same time
+	* at least resource ref or resource spec is defined
+*/
```

Comment:
(not very important: ) i think that `/* */` syntax for docstrings is pretty uncommon, usually I've seen `//` only :thinking: